solidity/contracts/BondedECDSAKeepVendor.sol

```diff
@@ -1,68 +1,42 @@
 
 /// @title Proxy contract for Bonded ECDSA Keep vendor.
-contract BondedECDSAKeepVendor is Proxy {
+contract BondedECDSAKeepVendor is Proxy, UpgradableProxyStorage {
     using SafeMath for uint256;
 
-    /// @dev Storage slot with the admin of the contract.
-    /// This is the keccak-256 hash of "eip1967.proxy.admin" subtracted by 1, and is
-    /// validated in the constructor.
-    bytes32 internal constant ADMIN_SLOT = 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
-
-    /// @dev Storage slot with the address of the current implementation.
-    /// This is the keccak-256 hash of "eip1967.proxy.implementation" subtracted by 1, and is
-    /// validated in the constructor.
-    bytes32 internal constant IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
-
-    /// @dev Storage slot with the upgrade time delay. Upgrade time delay defines a
-    /// period for implementation upgrade.
-    /// This is the keccak-256 hash of "network.keep.bondedecdsavendor.proxy.upgradeTimeDelay"
-    /// subtracted by 1, and is validated in the constructor.
-    bytes32 internal constant UPGRADE_TIME_DELAY_SLOT = 0x3ca583dafde9ce8bdb41fe825f85984a83b08ecf90ffaccbc4b049e8d8703563;
-
-    /// @dev Storage slot with the new implementation address.
-    /// This is the keccak-256 hash of "network.keep.bondedecdsavendor.proxy.upgradeImplementation"
-    /// subtracted by 1, and is validated in the constructor.
-    bytes32 internal constant UPGRADE_IMPLEMENTATION_SLOT = 0x4e06287250f0fdd90b4a096f346c06d4e706d470a14747ab56a0156d48a6883f;
-
-    /// @dev Storage slot with the implementation address upgrade initiation.
-    /// This is the keccak-256 hash of "network.keep.bondedecdsavendor.proxy.upgradeInitiatedTimestamp"
-    /// subtracted by 1, and is validated in the constructor.
-    bytes32 internal constant UPGRADE_INIT_TIMESTAMP_SLOT = 0x0816e8d9eeb2554df0d0b7edc58e2d957e6ce18adf92c138b50dd78a420bebaf;
-
-    event UpgradeStarted(address implementation, uint256 timestamp);
-    event UpgradeCompleted(address implementation);
-
-    constructor(address _implementation, bytes memory _data) public {
-        assertSlot(IMPLEMENTATION_SLOT, "eip1967.proxy.implementation");
-        assertSlot(ADMIN_SLOT, "eip1967.proxy.admin");
-        assertSlot(
-            UPGRADE_TIME_DELAY_SLOT,
-            "network.keep.bondedecdsavendor.proxy.upgradeTimeDelay"
-        );
-        assertSlot(
-            UPGRADE_IMPLEMENTATION_SLOT,
-            "network.keep.bondedecdsavendor.proxy.upgradeImplementation"
-        );
-        assertSlot(
-            UPGRADE_INIT_TIMESTAMP_SLOT,
-            "network.keep.bondedecdsavendor.proxy.upgradeInitiatedTimestamp"
-        );
+    event UpgradeStarted(
+        string version,
+        address implementation,
+        bytes initialization,
+        uint256 timestamp
+    );
+    event UpgradeCompleted(string version, address implementation);
```

Comment:
Should we emit address of the implementation again here? I think emitting just the version should be enough. We emitted it for `UpgradeStarted` event.